examples/servo/src/on_events.rs

```diff
@@ -3,146 +3,165 @@
-        units::{DevicePixel, DevicePoint},
-    },
+    InputEvent, MouseButton, MouseButtonAction, MouseButtonEvent, MouseMoveEvent, Scroll, Theme,
+    TouchEvent, TouchEventType, TouchId, WebViewPoint,
+    webrender_api::units::{DevicePixel, DevicePoint, DeviceVector2D},
 };
-use slint::{ComponentHandle, platform::PointerEventButton};
-use url::Url;
-use winit::dpi::PhysicalSize;
 
 use crate::{
     WebviewLogic,
     adapter::{SlintServoAdapter, upgrade_adapter},
 };
 
 pub fn on_app_callbacks(adapter: Rc<SlintServoAdapter>) {
+    on_url(adapter.clone());
+    on_theme(adapter.clone());
     on_resize(adapter.clone());
-    on_buttons(adapter.clone());
     on_scroll(adapter.clone());
+    on_buttons(adapter.clone());
     on_pointer(adapter.clone());
 }
 
-fn on_buttons(adapter: Rc<SlintServoAdapter>) {
+fn on_url(adapter: Rc<SlintServoAdapter>) {
     let app = adapter.app();
 
     let adapter_weak = Rc::downgrade(&adapter);
-    app.on_back(move || {
-        let adapter = upgrade_adapter(&adapter_weak);
 
+    app.global::<WebviewLogic>().on_loadUrl(move |url| {
+        let adapter = upgrade_adapter(&adapter_weak);
         let webview = adapter.webview();
-
-        webview.go_back(1);
+        let url = Url::parse(url.as_str()).expect("Failed to parse url");
+        webview.load(url);
     });
+}
+
+fn on_theme(adapter: Rc<SlintServoAdapter>) {
+    let app = adapter.app();
 
     let adapter_weak = Rc::downgrade(&adapter);
-    app.on_forward(move || {
+
+    app.global::<WebviewLogic>().on_theme(move |color_scheme| {
+        let theme = if color_scheme == ColorScheme::Dark { Theme::Dark } else { Theme::Light };
+
         let adapter = upgrade_adapter(&adapter_weak);
 
         let webview = adapter.webview();
 
-        webview.go_forward(1);
+        // Theme not updating its the issue with servo itself until mouse move over it
+        // https://github.com/servo/servo/issues/40268
+        webview.notify_theme_change(theme);
     });
+}
+
+fn on_resize(adapter: Rc<SlintServoAdapter>) {
+    let app = adapter.app();
 
     let adapter_weak = Rc::downgrade(&adapter);
-    app.on_reload(move || {
+    app.global::<WebviewLogic>().on_resize(move |width, height| {
         let adapter = upgrade_adapter(&adapter_weak);
 
         let webview = adapter.webview();
 
-        webview.reload();
-    });
+        let scale_factor = adapter.scale_factor();
 
-    let adpater_weak = Rc::downgrade(&adapter);
-    app.on_go(move |url| {
-        let adapter = upgrade_adapter(&adpater_weak);
-        let webview = adapter.webview();
-        let url = Url::parse(url.as_str()).expect("Failed to parse url");
-        webview.load(url);
+        let size = Size2D::new(width, height) * scale_factor;
+
+        let physical_size = PhysicalSize::new(size.width as u32, size.height as u32);
+
+        let rect: Box2D<f32, DevicePixel> = Box2D::from_origin_and_size(Point2D::origin(), size);
+
+        webview.move_resize(rect);
```

Comment:
I'm curious: Why is `move_resize()` called here? Is that call done to basically "scroll" to the top of the new page?